src/rbtree.rs

```diff
@@ -1180,7 +1180,7 @@ where
     /// This returns None if the cursor is currently pointing to the null
     /// object.
     #[inline]
-    pub fn get(&self) -> Option<&<A::PointerOps as PointerOps>::Value> {
+    pub fn get(&self) -> Option<&'a <A::PointerOps as PointerOps>::Value> {
```

Comment:
sorry, but I cannot find any mention of `get_mut`, what is it?

Reply:
Sorry I got confused. A better example would be that you could call `remove` after `get` without the lifetimes conflicting. This is incorrect since `remove` returns a uniquely owned pointer to the object (e.g. `Box`).

Reply:
oh I see, thanks

By any chance do you see any possible solution for my usecase? I've noticed `.as_cursor` method for `CursorMut` to get `Cursor`, but lifetime bounds are as strict as original `CursorMut`'s; what I want to achieve is to have a reference to just inserted item to live as long as the tree rather than returned `CursorMut` after `.insert` of a tree without starting a tree traversal over again.

Reply:
I think it might be possible to do this by *consuming* the `CursorMut`:
```rust
fn into_ref(self) -> Option<&'a <A::PointerOps as PointerOps>::Value> { ... }
```

This avoids the unsoundness issue since it prevents further use of the `CursorMut`.

Reply:
Thank you, I made a PR for this: https://github.com/Amanieu/intrusive-rs/pull/72